src/components/HeaderComponent/index.js

```diff
@@ -0,0 +1,266 @@
+import React, { useState } from 'react';
+import {
+  StyleSheet,
+  View,
+  Image,
+  TouchableOpacity,
+  Dimensions,
+  SafeAreaView,
+  Text,
+  Linking
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  MaterialIcons,
+  Ionicons
+} from '@expo/vector-icons';
+import Icon from 'react-native-remix-icon';
+import Spinner from 'react-native-loading-spinner-overlay';
+import { watchAd, getAdRedirectUrl } from '../../services/ad.services';
+import { obtainStickers } from '../../services/sticker.services';
+import { ModalPopup } from '../ModalPopup';
+import StickerTemplate from '../StickerTemplate';
+
+// Imagenes del Modal
+import logo from '../../../assets/appAssets/logo.png';
+import botonX from '../../../assets/appAssets/x.png';
+import sobre from '../../../assets/appAssets/sobre.png';
+
+const { width, height } = Dimensions.get('window');
+
+export default function HeaderComponent() {
+  const [loading, setLoading] = useState(false);
+  const [visibleObtener, setVisibleObtener] = useState(false);
+  const [visibleAnuncio, setVisibleAnuncio] = useState(false);
+  const [visibleStickers, setVisibleStickers] = useState(false);
+  const [ad, setAd] = useState(null);
+  const [obtainedStickers, setObtainedStickers] = useState([]);
+
+  const { token } = useSelector(state => state.auth);
+
+  const onClaimClick = async () => {
+    setLoading(true);
+    try {
+      const ad = await watchAd();
+      setAd(ad);
+      setVisibleAnuncio(true);
+    } catch (error) {
+      alert(error.message);
+    } finally {
+      setLoading(false);
+      setVisibleAnuncio(true);
+    }
```

Comment:
```suggestion
    try {
      const ad = await watchAd();
      setAd(ad);
    } catch (error) {
      alert(error.message);
    } finally {
      setLoading(false);
      setVisibleAnuncio(true);
    }
```
No repitas en el "try" algo que de todas formas pasara en el finally